### PR TITLE
[486] Change artifact name to incubator-xtable and Add deploy plugin to mvn

### PR DESCRIPTION
**Pull request by vinishjail97**
## *Important Read*
- *Please ensure the GitHub issue is mentioned at the beginning of the PR*

## What is the purpose of the pull request

https://github.com/apache/incubator-xtable/issues/486 
According to apache guidelines, we need to prefix the "incubator" prefix for all artifacts deployed to maven. This is how the bundled jar will look like in maven central repository when uploaded.
```
<dependency>
<groupId>org.apache.xtable</groupId>
<artifactId>incubator-xtable-utilities</artifactId>
<version>0.1.0-rc1</version>
<classifier>bundled</classifier>
</dependency>
```

## Brief change log

*(for example:)*
- *Added the mvn release and deploy plugins*
- *Made few other changes for generating java docs when lombok builders are present*

## Verify this pull request

*(Please pick either of the following options)*

This pull request is a trivial rework / code cleanup without any test coverage.

Validated the below command by uploading the rc-1 artifacts to staging repository. 
`mvn deploy -Prelease -DskipTests  -DdeployArtifacts=true`

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -770,4 +822,78 @@
...
+                            <execution>
+                                <id>attach-javadocs</id>
+                                <goals>
+                                    <goal>jar</goal>
+                                </goals>
+                            </execution>
+                        </executions>
+                        <configuration>
+                            <doclint>none</doclint>
+                            <source>11</source>
```
Comment (the-other-tim-brown) [inline pom.xml:877]: We're generating the jars with a java 8 target. Does this need to be 8?

Comment (vinishjail97) [reply pom.xml:877]: The plugin was failing to generate docs for few classes. 
```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-javadoc-plugin:3.8.0:jar (attach-javadocs) on project incubator-xtable-api: MavenReportException: Error while generating Javadoc: 
[ERROR] Exit code: 1
[ERROR] /Users/vinishreddy/helm/configs/incubator-xtable/xtable-api/target/delombok/org/apache/xtable/spi/extractor/ExtractFromSource.java:38: error: method references are not supported in -source 7
[ERROR]     Iterator<TableChange> tableChangeIterator = commitsBacklog.getCommitsToProcess().stream().map(conversionSource::getTableChangeForCommit).iterator();
[ERROR]                                                                                                                     ^
[ERROR]   (use -source 8 or higher to enable method references)
[ERROR] /Users/vinishreddy/helm/configs/incubator-xtable/xtable-api/target/delombok/org/apache/xtable/spi/sync/TableFormatSync.java:64: error: lambda expressions are not supported in -source 7
[ERROR]         results.put(conversionTarget.getTableFormat(), getSyncResult(conversionTarget, SyncMode.FULL, internalTable, target -> target.syncFilesForSnapshot(snapshot.getPartitionedDataFiles()), startTime, snapshot.getPendingCommits()));
[ERROR]                                                                                                                             ^
[ERROR]   (use -source 8 or higher to enable lambda expressions)
[ERROR] /Users/vinishreddy/helm/configs/incubator-xtable/xtable-api/target/delombok/org/apache/xtable/spi/sync/TableFormatSync.java:88: error: method references are not supported in -source 7
[ERROR]       }).map(Map.Entry::getKey).collect(Collectors.toList());
[ERROR]                         ^
[ERROR]   (use -source 8 or higher to enable method references)
[ERROR] /Users/vinishreddy/helm/configs/incubator-xtable/xtable-api/target/delombok/org/apache/xtable/model/storage/DataFilesDiff.java:39: error: method references are not supported in -source 7
[ERROR]     Map<String, InternalDataFile> targetPaths = target.stream().collect(Collectors.toMap(InternalDataFile::getPhysicalPath, Function.identity()));
[ERROR]                                                                                                            ^
[ERROR]   (use -source 8 or higher to enable method references)
[ERROR] /Users/vinishreddy/helm/configs/incubator-xtable/xtable-api/target/delombok/org/apache/xtable/model/storage/FilesDiff.java:59: error: lambda expressions are not supported in -source 7
[ERROR]     latestFiles.forEach((key, value) -> {
[ERROR]                                      ^
[ERROR]   (use -source 8 or higher to enable lambda expressions)
[ERROR] /Users/vinishreddy/helm/configs/incubator-xtable/xtable-api/target/delombok/org/apache/xtable/model/storage/FilesDiff.java:81: error: method references are not supported in -source 7
[ERROR]     Map<String, InternalDataFile> latestFiles = latestFileGroups.stream().flatMap(group -> group.getFiles().stream()).collect(Collectors.toMap(InternalDataFile::getPhysicalPath, Function.identity()));
[ERROR]                                                                                                                                                                  ^
[ERROR]   (use -source 8 or higher to enable method references)
[ERROR] /Users/vinishreddy/helm/configs/incubator-xtable/xtable-api/target/delombok/org/apache/xtable/model/storage/PartitionFileGroup.java:39: error: method references are not supported in -source 7
[ERROR]     Map<List<PartitionValue>, List<InternalDataFile>> filesGrouped = files.collect(Collectors.groupingBy(InternalDataFile::getPartitionValues));
[ERROR]                                                                                                                            ^
[ERROR]   (use -source 8 or higher to enable method references)
[ERROR] /Users/vinishreddy/helm/configs/incubator-xtable/xtable-api/target/delombok/org/apache/xtable/model/storage/PartitionFileGroup.java:40: error: lambda expressions are not supported in -source 7
[ERROR]     return filesGrouped.entrySet().stream().map(entry -> PartitionFileGroup.builder().partitionValues(entry.getKey()).files(entry.getValue()).build()).collect(Collectors.toList());
[ERROR]                                                       ^
[ERROR]   (use -source 8 or higher to enable lambda expressions)
[ERROR] Command line was: /opt/homebrew/Cellar/openjdk@11/11.0.22/libexec/openjdk.jdk/Contents/Home/bin/javadoc -J-Duser.language= -J-Duser.country= @options @argfile
[ERROR] 
[ERROR] Refer to the generated Javadoc files in '/Users/vinishreddy/helm/configs/incubator-xtable/xtable-api/target/apidocs' dir.

```

Comment (the-other-tim-brown) [reply pom.xml:877]: Did you try with 1.8 or 8? Sometimes these tools require 1.8

Comment (vinishjail97) [reply pom.xml:877]: 1.8 works.

Comment (vinishjail97) [reply pom.xml]: Verified it through the mvn deploy command.

**Comment by vinishjail97** [comment]
Squashed everything into a single commit.

**Comment by jcamachor** [comment]
Sorry for the late response, I was OOF. Thanks, @vinishjail97, for putting the PR together. Could you share the ASF guidelines you are referring to? AFAIK the common practice is adding the `-incubating` suffix to the release version until the project graduates to TLP, rather than changing the artifact names (which makes sense so any consumer can update the version without changing the artifact ID all over). From a quick look at other incubating projects, that seem to be the practice that they are following. Also cc'ing @zabetak and @jbonofre in case they have any insights about this.

**Comment by vinishjail97** [comment]
The [Incubator policies](https://incubator.apache.org/incubation/Incubation_Policy.html#Releases) applies two additional constraints to podlings for their releases. They are repeated here for clarity only. - Release artifacts must include incubating in the final file name - Release artifacts must include one of two [disclaimers](https://incubator.apache.org/guides/branding.html#disclaimers)


I was refereeing to this guidelines, reading it again looks like you may be right. I misunderstood this sentence. 
`Release artifacts must include incubating in the final file name `

**Comment by jcamachor** [comment]
Yes, looking at the artifacts of other incubating projects, version looks like `<version>-incubating`.

**Pull request merged**